### PR TITLE
Maintain content type for data parts in multipart requests

### PR DESCRIPTION
**Pull request by eloipoch**
Any changes required or improvements proposed are welcome.

This PR fixes the issue #1581

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@eloipoch Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1614)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1614) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@eloipoch Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1614)!

**Comment by eloipoch** [comment]
Of course 👌

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/util/RequestContentDataExtractor.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    private static MultiValueMap<String, Object> extractFromRequest(HttpServletRequest request) throws IOException {
+        MultiValueMap<String, Object> builder     = new LinkedMultiValueMap<>();
+        Set<String>                   queryParams = findQueryParams(request);
+
+        for (Entry<String, String[]> entry : request.getParameterMap().entrySet()) {
+            String key = entry.getKey();
+
+            if (!queryParams.contains(key)) {
+                for (String value : entry.getValue()) {
+                    builder.add(entry.getKey(), value);
```
Comment (benoit-lateltin) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/util/RequestContentDataExtractor.java]: `entry.getKey()` could be replaced by `key`

Comment (eloipoch) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/util/RequestContentDataExtractor.java]: Thanks!

**Comment by eloipoch** [comment]
I've done a refactor in the test class (LocalServerPort, injected test rest template, and a couple of helper methods).
If it is not welcome for any reason I will revert the commit without hesitation

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/pre/FormBodyWrapperFilter.java [hunk 2]
```diff
@@ -184,36 +178,9 @@ public ServletInputStream getInputStream() throws IOException {
...
-							HttpEntity<Resource> entity = new HttpEntity<Resource>(
-									new InputStreamResource(file.getInputStream()),
-									headers);
-							builder.add(parts.getKey(), entity);
-						}
-					}
-				}
-				FormHttpOutputMessage data = new FormHttpOutputMessage();
+				MultiValueMap<String, Object> builder = RequestContentDataExtractor.extract(this.request);
+				FormHttpOutputMessage         data    = new FormHttpOutputMessage();
```
Comment (ryanjbaxter) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/pre/FormBodyWrapperFilter.java]: some spacing issues here

**Pull request merged**